### DISCUSSION TITLE
Agent stage optimization algorithm: EGM

### DISCUSSION THREAD

**Discussion by sbenthall** [econ-ark/HARK#1652] [Ideas]
Agent stage optimization algorithm. EGM

-  for a single stage
-  looping through a fully specified model (maybe multi-stage)

In my view, this would be the right place to generalize the `ConsLabelModel` from
https://github.com/econ-ark/HARK/blob/master/HARK/ConsumptionSaving/ConsLabeledModel.py

... and do a rewrite using a more general model representation (such as the one in `model.py`). @alanlujan91 I wonder if you agree.

**Comment by sbenthall** [comment]
A general algorithm is implemented in https://github.com/econ-ark/HARK/pull/1283, but it uses a slightly different model representation. So it needs to be ported over. (Or rewritten from scratch)

**Comment by sbenthall** [comment]
Also, the algorithm from [PensionRetirementModel.ipynb](https://github.com/econ-ark/OverARK/blob/master/Development/nohablo/PensionRetirementModel.ipynb)

 https://github.com/econ-ark/OverARK/tree/master/Development/nohablo

**Comment by mnwhite** [comment]
Moving this (and a lot of other "big picture" items) to Discussions. Many of the things I'm moving are going to be implemented in the next big Econ-ARK project, but not necessarily in HARK. I'd like to keep the Issues list limited to bugs and reasonably feasible feature requests, rather than structural changes.
